Sources/TestingMacros/Support/AttributeDiscovery.swift

```diff
@@ -180,6 +180,22 @@ struct AttributeInfo {
         ArrayElementSyntax(expression: traitExpr)
       }
     }))
+
+    var otherArguments = self.otherArguments
+
+    // Any arguments of the test declaration macro which specify test arguments
+    // need to be wrapped a closure so they may be evaluated lazily by the
+    // testing library at runtime. If any such arguments are present, they will
+    // begin with a labeled argument named `arguments:` and include all
+    // subsequent unlabeled arguments.
+    if let argumentsIndex = otherArguments.firstIndex(where: { $0.label?.tokenKind == .identifier("arguments") }) {
+      for index in argumentsIndex ..< otherArguments.endIndex {
```

Comment:
I think you could do:

```swift
arguments[argumentsIndex...] = arguments[argumentsIndex...].map { argument in
  ...
}
```

Reply:
Definitely open to this but I can't get this suggestion to compile, even if I drop `.lazy` or surround it with `Array(...)`

Reply:
It's not critical, just a bit closer to our house style of _use all the monads!_. :) No worries.